### ISSUE TITLE
container image id (with hash no tag)

### ISSUE THREAD

**Issue by yonixw** [google/cadvisor#2635]
Hi, I cannot find how to report the docker container image id (the sha256) and not the tag.
Is it by design? Can I config it anywhere?

My use case is to monitor the exact versions across multiple deployments and monitor update processes.

Thank in advance.

**Comment by dashpole** [comment]
Yeah, it looks like we only support whatever the container runtime returns to us, which appears to be whatever you set in the pod spec: https://github.com/google/cadvisor/blob/master/container/docker/handler.go#L218.  

I would suggest getting this from kube-state-metrics using `kube_pod_container_info`: https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/pod.go#L421

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
